model/CompatibilityCheckerService.php

```diff
@@ -63,43 +68,64 @@ function __construct()
      *  - return 2 if not tested
      * @return int
      */
-    public function isCompatibleConfig()
+    public function isCompatibleConfig($check = 3)
     {
         $browserVersion = explode('.', $this->browserVersion);
+        $rules = array_filter($this->compatibility, function ($val) {
+            return $val->os === $this->os;
+        });
+
+        foreach ($rules as $rule) {
+            $validOs = true;
+            $validBrowser = false;
+
+            if ($rule->osVersion !== "") {
+                $validOs = $this->versionCompare($this->osVersion, $rule->osVersion) === 0;
+            }
+
+            if ($check === self::CHECK_OS && $validOs && $rule->compatible) {
+                return 1;
+            }
+
 
-        $osVersion = explode('.', $this->osVersion);
-        foreach ($this->compatibility as $rule) {
-            //os name
-            if ($rule->os === $this->os) {
-                //os Version
-                $validOs = true;
-                if ($rule->osVersion !== "") {
-                    foreach (explode('.', $rule->osVersion) as $key => $version) {
-                        if (!isset($osVersion[$key]) || $osVersion[$key] !== $version) {
-                            $validOs = false;
-                        }
-                    }
-                }
-
-                //browser validation
-                if (empty($rule->versions)) {
-                    // all versions work
-                    $isValid = true;
-                } else {
-                    // it is valid if the version is in the array
-                    // OR if the browser is chrome or firefox and it is a newer version than those in the array
-                    $isValid = in_array($browserVersion[0], $rule->versions)
-                        || (in_array($rule->browser,
-                                array('Chrome', 'Firefox')) && $browserVersion[0] > max($rule->versions));
-                }
-
-                if ($validOs && ($rule->browser === ""
-                        || $rule->browser === $this->browser && $isValid)
-                ) {
-                    return $rule->compatible;
-                }
+            //browser validation
+            if ($rule->browser == $this->browser && empty($rule->versions)) {
+                // all versions work
+                $validBrowser = true;
+            } elseif($rule->browser == $this->browser) {
+                // it is valid if the version is in the array
+                // OR if the browser is chrome or firefox and it is a newer version than those in the array
+                $validBrowser = in_array($browserVersion[0], $rule->versions)
+                    || (in_array($rule->browser, ['Chrome', 'Firefox']) && $browserVersion[0] > max($rule->versions));
+            } else {
+                $validBrowser = false;
+            }
+
```

Comment:
but we return `$rule->compatible` in case if entry for current os and browser found:
https://github.com/oat-sa/extension-tao-clientdiag/pull/131/files#diff-987d9da56d7cdf9988cd2a44518c7351R105